nixio/section.py

```diff
@@ -75,39 +79,120 @@ def create_section(self, name, type_):
         if name in sections:
             raise exceptions.DuplicateName("create_section")
-        sec = Section._create_new(self, sections, name, type_)
+        sec = Section._create_new(self, sections, name, type_, oid)
         sec._sec_parent = self
         return sec
 
     # Property
-    def create_property(self, name, values):
+    def create_property(self, name, values_or_dtype, oid=None):
         """
         Add a new property to the section.
 
         :param name: The name of the property to create.
         :type name: str
-        :param values: The values of the property.
-        :type values: list of Value
+        :param values_or_dtype: The values of the property or a valid DataType.
+        :type values_or_dtype: list of values or a DataType
+        :param oid: object id, UUID string as specified in RFC 4122. If no id is provided,
+                   an id will be generated and assigned.
+        :type oid: str
 
         :returns: The newly created property.
         :rtype: Property
         """
+        vals = values_or_dtype
+
         properties = self._h5group.open_group("properties", True)
         if name in properties:
             raise exceptions.DuplicateName("create_property")
-        if isinstance(values, type):
-            dtype = values
-            values = []
+
+        # Handle handed in DataType
+        if isinstance(vals, type):
+            dtype = vals
+            vals = []
+
+        # In case of values, make sure boolean value 'False' gets through as well,
+        # but ensure that empty values are not allowed, we need a DataType.
+        elif vals is None or (isinstance(vals, Sequence) and not vals):
+            raise TypeError("Please provide either a non empty value or a DataType.")
+
         else:
-            if isinstance(values, Sequence):
-                dtype = values[0].data_type
+            # Make sure all values are of the same data type
+            single_val = vals
+            if isinstance(vals, Sequence):
+                single_val = vals[0]
             else:
-                dtype = values.data_type
-                values = [values]
-        prop = Property._create_new(self, properties, name, dtype)
-        prop.values = values
+                # Make sure the data will always be created with an array.
+                vals = [vals]
+
+            # Will raise an error, if the datatype of the first value is not valid.
+            dtype = DataType.get_dtype(single_val)
+
+            # Check all values for data type consistency to ensure clean value add.
+            # Will raise an exception otherwise.
+            for v in vals:
+                if DataType.get_dtype(v) != dtype:
+                    raise exceptions.InvalidAttrType(single_val, DataType.get_dtype(v))
+
+        prop = Property._create_new(self, properties, name, dtype, oid)
+        prop.values = vals
+
         return prop
 
+    def has_property_by_name(self, name):
```

Comment:
These methods (`has_property_by_name`, `get_property_by_name`, etc) are the old-style ProxyList methods and aren't used anymore. This functionality is now handled by the Container class which lives on the Section as `_properties` and is exposed through `props`.